### PR TITLE
Add test targets to Makefile

### PR DESCRIPTION
**Pull request by jose-bigio**
Signed-off-by: jose-bigio <jose.bigio@docker.com>

**What this PR does / why we need it**:
Adds a Make target so the tests can be easily run.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -97,6 +97,13 @@ all-binaries: azure-cnm-plugin azure-cni-plugin azure-cns
 clean:
 	rm -rf $(OUTPUT_DIR)
 
+.PHONY: test
```
[the remark this answers is not in the document]
Comment (jose-bigio) [reply Makefile]: @tamilmani1989  I modified the Make target let me know if this work.

[the remark this answers is not in the document]
Comment (jose-bigio) [reply Makefile]: @tamilmani1989 wanted to follow up regarding the change.

[the remark this answers is not in the document]
Comment (tamilmani1989) [reply Makefile]: @jose-bigio Did you test from your side first? When I do make test from root directory(azure-container-networking), its failing.

[the remark this answers is not in the document]
Comment (jose-bigio) [reply Makefile]: Here's what I get when I run it: 

```
~/go/src/github.com/Azure/azure-container-networking test_containerized*
❯ make test
cd log && go test
# github.com/Azure/azure-container-networking/log
./logger.go:64:8: logger.SetTarget undefined (type Logger has no field or method SetTarget)
./logger.go:106:9: undefined: platform.LogPath
./logger.go:116:17: undefined: platform.LogPath
./logger.go:155:9: logger.SetTarget undefined (type *Logger has no field or method SetTarget)
./stdapi.go:19:15: stdLog.SetTarget undefined (type *Logger has no field or method SetTarget)
FAIL	github.com/Azure/azure-container-networking/log [build failed]
make: *** [test-log] Error 2
```

but I get these errors for other tests when I run them individually (make target would be doing the same thing I think): 
```
cd ipam && go test
# github.com/Azure/azure-container-networking/log
../log/logger.go:64:8: logger.SetTarget undefined (type Logger has no field or method SetTarget)
../log/logger.go:106:9: undefined: platform.LogPath
../log/logger.go:116:17: undefined: platform.LogPath
../log/logger.go:155:9: logger.SetTarget undefined (type *Logger has no field or method SetTarget)
../log/stdapi.go:19:15: stdLog.SetTarget undefined (type *Logger has no field or method SetTarget)
FAIL	github.com/Azure/azure-container-networking/ipam [build failed]
```

```
cd cni && go test
# github.com/Azure/azure-container-networking/log
../log/logger.go:64:8: logger.SetTarget undefined (type Logger has no field or method SetTarget)
../log/logger.go:106:9: undefined: platform.LogPath
../log/logger.go:116:17: undefined: platform.LogPath
../log/logger.go:155:9: logger.SetTarget undefined (type *Logger has no field or method SetTarget)
../log/stdapi.go:19:15: stdLog.SetTarget undefined (type *Logger has no field or method SetTarget)
```

```
cd cnm && go test
# github.com/Azure/azure-container-networking/log
../log/logger.go:64:8: logger.SetTarget undefined (type Logger has no field or method SetTarget)
../log/logger.go:106:9: undefined: platform.LogPath
../log/logger.go:116:17: undefined: platform.LogPath
../log/logger.go:155:9: logger.SetTarget undefined (type *Logger has no field or method SetTarget)
../log/stdapi.go:19:15: stdLog.SetTarget undefined (type *Logger has no field or method SetTarget
```

```
cd netlink && go test
# github.com/Azure/azure-container-networking/log
../log/logger.go:64:8: logger.SetTarget undefined (type Logger has no field or method SetTarget)
../log/logger.go:106:9: undefined: platform.LogPath
../log/logger.go:116:17: undefined: platform.LogPath
../log/logger.go:155:9: logger.SetTarget undefined (type *Logger has no field or method SetTarget)
../log/stdapi.go:19:15: stdLog.SetTarget undefined (type *Logger has no field or method SetTarget)
FAIL	github.com/Azure/azure-container-networking/netlink [build failed]
```

```
cd cns && go test
# github.com/Azure/azure-container-networking/log
../log/logger.go:64:8: logger.SetTarget undefined (type Logger has no field or method SetTarget)
../log/logger.go:106:9: undefined: platform.LogPath
../log/logger.go:116:17: undefined: platform.LogPath
../log/logger.go:155:9: logger.SetTarget undefined (type *Logger has no field or method SetTarget)
../log/stdapi.go:19:15: stdLog.SetTarget undefined (type *Logger has no field or method SetTarget)
FAIL	github.com/Azure/azure-container-networking/netlink [build failed]
```

```
cd telemetry && go test
# github.com/Azure/azure-container-networking/log
../log/logger.go:64:8: logger.SetTarget undefined (type Logger has no field or method SetTarget)
../log/logger.go:106:9: undefined: platform.LogPath
../log/logger.go:116:17: undefined: platform.LogPath
../log/logger.go:155:9: logger.SetTarget undefined (type *Logger has no field or method SetTarget)
../log/stdapi.go:19:15: stdLog.SetTarget undefined (type *Logger has no field or method SetTarget)
FAIL	github.com/Azure/azure-container-networking/telemetry [build failed]
```

So yeah the tests are failing, but I don't know if they are failing, due to how they are run.
It seems as if the tests don't even compile though.

**Pull request closed without merging**